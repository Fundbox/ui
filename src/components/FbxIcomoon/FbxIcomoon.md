# FbxIcomoon
An icomoon component.
Pass in the name of the icon.
