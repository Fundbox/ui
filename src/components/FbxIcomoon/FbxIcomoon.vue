<template>
  <span class="icon-wrapper" :class="classes" :style="styles"></span>
</template>

<script>
export default {
  name: 'FbxIcomoon',
  props: {
    name: {
      type: String,
      required: true,
    },
    background: {
      type: String,
      default: undefined,
    },
    color: {
      type: String,
      default: undefined,
    },
    round: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    classes() {
      return {
        [`fbx-icon-${this.name}`]: true,
        round: this.round,
        styled: this.round || this.background,
      }
    },
    styles() {
      return {
        background: this.background,
        color: this.color,
      }
    }
  }
}
</script>

<style lang="scss" scoped>
@import '../../styles/fonts/icomoon/style.css';
@import '../../styles/utils/utilities.scss';

$icon-size: 2.5em;

.icon-wrapper{
  display: inline-block;
  position: relative;

  &.styled {
    height: $icon-size;
    width: $icon-size;
  }

  &.round {
    border-radius: 50%;
  }

  &:before {
    background-color: inherit;
    color: inherit;
    @extend .absolute-center;
    // Icomoon icons are not dead centered.
    transform: translate(-47%, -56%);
    font-size: inherit;
  }
}
</style>
