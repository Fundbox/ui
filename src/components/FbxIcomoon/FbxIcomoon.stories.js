import { storiesOf } from '@storybook/vue'
import { withInfo } from 'storybook-addon-vue-info'

import FbxIcomoon from './FbxIcomoon.vue'
import summary from './FbxIcomoon.md'

const stories = storiesOf('Components/Icomoon', module)

stories.add('default', withInfo({ summary })(() => ({
  components: { FbxIcomoon },
  template: `<fbx-icomoon name="lock"></fbx-icomoon>`
})))

stories.add('with background styles', withInfo({ summary })(() => ({
  components: { FbxIcomoon },
  template: `<fbx-icomoon name="lock" background="#858585" color="white" round></fbx-icomoon>`
})))

stories.add('default', withInfo({ summary })(() => ({
  components: { FbxIcomoon },
  template: `<fbx-icomoon name="lock"></fbx-icomoon>`
})))
