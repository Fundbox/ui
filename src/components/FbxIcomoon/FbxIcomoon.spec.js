import { shallowMount } from '@vue/test-utils'
import FbxIcomoon from './FbxIcomoon.vue'

describe('Components/FbxIcomoon', () => {
  describe('logic', () => {
    it('test for required param', () => {
      let spy = jest.spyOn(console, 'error').mockImplementation(() => null)
      shallowMount(FbxIcomoon)
      expect(spy).toHaveBeenCalledWith(expect.stringContaining(`[Vue warn]: Missing required prop: "name"`))
      spy.mockRestore()
    })
  })

  describe('snapshots', () => {
    it('renders the default correctly', () => {
      const wrapper = shallowMount(FbxIcomoon, {
        propsData: {
          name: 'lock',
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('renders background color and round', () => {
      const wrapper = shallowMount(FbxIcomoon, {
        propsData: {
          name: 'lock',
          background: '#858585',
          color: 'white',
          round: true,
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
