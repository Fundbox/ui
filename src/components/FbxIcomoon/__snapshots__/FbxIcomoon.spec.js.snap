// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/FbxIcomoon snapshots renders background color and round 1`] = `<span class="icon-wrapper fbx-icon-lock round styled" style="background: rgb(133, 133, 133); color: white;"></span>`;

exports[`Components/FbxIcomoon snapshots renders the default correctly 1`] = `<span class="icon-wrapper fbx-icon-lock"></span>`;
